okhttp/src/main/java/okhttp3/ConnectionSpec.kt

```diff
@@ -183,8 +183,8 @@ class ConnectionSpec internal constructor(
   override fun hashCode(): Int {
```

Comment:
In `equals(other: Any?)`, it's good to remove `!!` from `if (this.isTls != that!!.isTls) return false`? 🤔 I don't think there's a chance that NEP can happen though (since we have a sanity check `if (other !is ConnectionSpec) return false`) at the top of the function.
```
override fun equals(other: Any?): Boolean {
  if (other !is ConnectionSpec) return false	
  if (other === this) return true

  val that = other as ConnectionSpec?
  if (this.isTls != that!!.isTls) return false
```

Reply:
The cast for that is wrong and shouldn't be nullable

Reply:
Ah, right. Removed the wrong cast in `equals(...)`. Thank you!